options.go

```diff
@@ -1,14 +1,39 @@
 package smt
 
-// Option is a function that configures SparseMerkleTrie.
-type Option func(*TrieSpec)
+import (
+	"hash"
+)
+
+// TrieSpecOption is a function that configures SparseMerkleTrie.
+type TrieSpecOption func(*TrieSpec)
 
 // WithPathHasher returns an Option that sets the PathHasher to the one provided
-func WithPathHasher(ph PathHasher) Option {
+func WithPathHasher(ph PathHasher) TrieSpecOption {
 	return func(ts *TrieSpec) { ts.ph = ph }
 }
 
 // WithValueHasher returns an Option that sets the ValueHasher to the one provided
-func WithValueHasher(vh ValueHasher) Option {
+func WithValueHasher(vh ValueHasher) TrieSpecOption {
 	return func(ts *TrieSpec) { ts.vh = vh }
 }
+
+// NoHasherSpec returns a new TrieSpec that has nil ValueHasher & PathHasher specs.
+// NB: This should only be used when values are already hashed and a path is
+// used instead of a key during proof verification. Otherwise, this will lead
+// double hashing and product an incorrect leaf digest, thereby invalidating
+// the proof.
+// TODO_TECHDEBT: Document better when/why this is needed.
+func NoHasherSpec(hasher hash.Hash, sumTrie bool) *TrieSpec {
```

Comment:
```suggestion
func noHasherSpec(hasher hash.Hash, sumTrie bool) *TrieSpec {
```
Do not export this, this is only needed in the validation logic for closest proofs and **should never** be used for a trie

Reply:
Removed!